### PR TITLE
Add support for Global styles

### PR DESCRIPTION
**Pull request by natemoo-re**
When encountering a `<style global>` element, we'll handle it properly and won't scope any elements in the same file.

Closes https://github.com/snowpackjs/astro/issues/1295

### CODE REVIEW

File: internal/transform/scope-css.go [hunk 1]
```diff
@@ -6,18 +6,29 @@ import (
...
 			continue
 		}
-		n.Attr = append(n.Attr, tycho.Attribute{
+		for _, attr := range n.Attr {
+			if attr.Key == "global" &&
+				(attr.Type == astro.EmptyAttribute) ||
+				(attr.Type == astro.ExpressionAttribute && attr.Val == "true") ||
+				(attr.Type == astro.QuotedAttribute && (attr.Val == "" || attr.Val == "true")) {
+				continue outer
+			}
```
Comment (natemoo-re) [inline internal/transform/scope-css.go:28]: Skips `<style global>` nodes from scoping, returns a flag that we shouldn't scope HTML elements

File: internal/transform/transform.go [hunk 2]
```diff
@@ -16,11 +16,11 @@ func Transform(doc *tycho.Node, opts TransformOptions) {
...
-		ScopeStyle(doc.Styles, opts)
-
-		walk(doc, func(n *tycho.Node) {
-			ScopeElement(n, opts)
-		})
+		if shouldScope := ScopeStyle(doc.Styles, opts); shouldScope {
+			walk(doc, func(n *tycho.Node) {
+				ScopeElement(n, opts)
+			})
+		}
```
Comment (natemoo-re) [inline internal/transform/transform.go:23]: Conditionally scope HTML elements, rather than always

File: internal/print-to-source.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+				buf.WriteString("=`" + strings.TrimSpace(attr.Val) + "`")
+			}
+		}
+		buf.WriteString(`>`)
+		for c := node.FirstChild; c != nil; c = c.NextSibling {
+			PrintToSource(buf, c)
+		}
+		buf.WriteString(fmt.Sprintf(`</%s>`, node.Data))
+	}
+}
```
Comment (natemoo-re) [inline internal/print-to-source.go:52]: Extracted this to a shared `internal/` file. Only used for tests, but I was getting an import cycle when I tried to put it in the `test_utils` package.

It's nice how straight-forward this is!

**Review by FredKSchott — APPROVED** [review]
This one could probably use an extra look from @drwpow but overall lgtm

File: internal/transform/scope-css.go [hunk 4]
```diff
@@ -6,18 +6,29 @@ import (
...
+outer:
 	for _, n := range styles {
 		if n.DataAtom != a.Style {
 			continue
 		}
-		n.Attr = append(n.Attr, tycho.Attribute{
+		for _, attr := range n.Attr {
+			if attr.Key == "global" &&
+				(attr.Type == astro.EmptyAttribute) ||
+				(attr.Type == astro.ExpressionAttribute && attr.Val == "true") ||
```
Comment (drwpow) [inline internal/transform/scope-css.go:25]: Ooh nice

**Review by drwpow — APPROVED** [review]
LGTM!

**Pull request merged**